### ISSUE TITLE
gen_snapshot hangs on corelib_2/regexp/pcre_test when obfuscation is enabled

### ISSUE THREAD

**Issue by sjindel-google** [dart-lang/sdk#37450]
See https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8908759399892586944/+/steps/test_results/0/logs/unapproved_failing_tests__logs_/0

This failure appears to have started on https://dart.googlesource.com/sdk/+/4babccd9885b4a50daf291278bb9a3d71a85cdcc, although that CL appears unrelated.

I've approved the timeout.

**Comment by sstrickl** [comment]
Yeah, its initial red was in the "Unapproved test failures" log only, not in a "New unapproved test failures" log, which generally means it was showing up in the "Ignored flaky test failures" log before this point.

Unfortunately, either the obfuscate bot isn't kept track of in the flakiness dashboard, or something else is causing these failures to not show up there, as that would be my first place to go to figure out approximately how long it's been showing up as flaky at all. Doing some quick scans of random points in the bot build history, it's been in the unapproved flaky test failure logs since at least [build 2150](https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8923549528193940720/+/steps/test_results/0/logs/ignored_flaky_test_failure_logs/0) (so since Jan 23(!)), but I don't have a firm CL causing it due to random sampling. (Also, I'm not sure when those ignored flaky test failure logs began to be kept, so it may have been a possible flake at other points I checked in earlier builds but not logged).

So this wasn't caused by my work on the RegExp engine in April (my initial concern), but it's still interesting that it has been flaky for so long and suddenly now has became consistent enough to throw up the red flags on the CI.

**Comment by athomas** [comment]
It looks like it started to fail consistently at some point. Before the red series of builds, the last green build showed that it had consistently gotten Timeouts for 99 builds:
https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8908904036129265216/+/steps/update_flakiness_information/0/logs/raw_io.output_text_flaky.json_/0#L47792_0

So on the seemingly unrelated failure it was no longer considered flaky, because it has consistently failed with a Timeout for 100 builds.

https://ci.chromium.org/p/dart/builders/ci.sandbox/vm-kernel-precomp-obfuscate-linux-release-x64/4022 is the last build a "Pass" was seen. All later builds got timeouts.

**Comment by sjindel-google** [comment]
This Timeout has appeared again:

https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8907070913779799936/+/steps/test_results/0/logs/unapproved_failing_tests__logs_/0
